test/integration/apiserver/flowcontrol/concurrency_util_test.go

```diff
@@ -0,0 +1,506 @@
+	testTime                           = 10 * time.Second
+)
+
+func setupWithAuthorizer(t testing.TB, maxReadonlyRequestsInFlight, maxMutatingRequestsInFlight int, authz authorizer.Authorizer) (*rest.Config, framework.TearDownFunc) {
+	_, kubeConfig, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			// Ensure all clients are allowed to send requests.
+			opts.Authorization.Modes = []string{"AlwaysAllow"}
+			opts.GenericServerRunOptions.MaxRequestsInFlight = maxReadonlyRequestsInFlight
+			opts.GenericServerRunOptions.MaxMutatingRequestsInFlight = maxMutatingRequestsInFlight
+		},
+		ModifyServerConfig: func(config *controlplane.Config) {
+			config.GenericConfig.Authorization.Authorizer = authz
+		},
+	})
+	return kubeConfig, tearDownFn
+}
+
+type SumAndCount struct {
+	Sum   float64
+	Count int
+}
+
+type plMetrics struct {
+	execSeconds    SumAndCount
+	seatUtil       SumAndCount
+	availableSeats int
+}
+
+// metricSnapshot maps from a priority level label to
+// a plMetrics struct containing APF metrics of interest
+type metricSnapshot map[string]plMetrics
+
+// Client request latency measurement
+type clientLatencyMeasurement struct {
+	SumAndCount
+	SumSq float64 // latency sum of squares
+	Mu    sync.Mutex
+}
+
+func (clm *clientLatencyMeasurement) reset() {
+	clm.Mu.Lock()
+	clm.Sum = 0
+	clm.Count = 0
+	clm.SumSq = 0
+	clm.Mu.Unlock()
+}
+
+func (clm *clientLatencyMeasurement) update(duration float64) {
+	clm.Mu.Lock()
+	clm.Count += 1
+	clm.Sum += duration
+	clm.SumSq += duration * duration
+	clm.Mu.Unlock()
+}
+
+func (clm *clientLatencyMeasurement) getStats() clientLatencyStats {
+	mean := clm.Sum / float64(clm.Count)
+	ss := clm.SumSq - mean*clm.Sum // reduced from ss := sumsq - 2*mean*sum + float64(count)*mean*mean
+	stdDev := math.Sqrt(ss / float64(clm.Count))
+	cv := stdDev / mean
+	return clientLatencyStats{mean: mean, stdDev: stdDev, cv: cv}
+}
+
+type clientLatencyStats struct {
+	mean   float64 // latency average
+	stdDev float64 // latency standard deviation
+	cv     float64 // latency coefficient of variation
+}
+
+type plMetricAvg struct {
+	reqExecution float64 // average request execution time
+	seatUtil     float64 // average seat utilization
+}
+
+func intervalMetricAvg(snapshots map[string]metricSnapshot, t0 string, t1 string, plLabel string) plMetricAvg {
+	plmT0 := snapshots[t0][plLabel]
+	plmT1 := snapshots[t1][plLabel]
+	return plMetricAvg{
+		reqExecution: (plmT1.execSeconds.Sum - plmT0.execSeconds.Sum) / float64(plmT1.execSeconds.Count-plmT0.execSeconds.Count),
+		seatUtil:     (plmT1.seatUtil.Sum - plmT0.seatUtil.Sum) / float64(plmT1.seatUtil.Count-plmT0.seatUtil.Count),
+	}
+}
+
+// This integration test checks the client-side expected concurrency and the server-side observed concurrency
+// to make sure that they are close within a small error bound and that the priority levels are isolated.
```

Comment:
I think a better high level description would focus on the isolation.  Perhaps something like the following.

Remember also the convention that the comment on something begins with its name.

```
// TestConcurrencyIsolation tests the concurrency isolation between priority levels.
// The test defines two priority levels for this purpose, and corresponding flow schemas.
// To one priority level, this test sends many more concurrent requests than the configuration
// allows to execute at once, while sending fewer than allowed to the other priority level.
// The primary check is that the low flow gets all the seats it wants, but modulated by
// recognizing that there are uncontrolled overheads in the system.
```